src/devices.js

```diff
@@ -54,22 +54,28 @@ export const requestXRDevice = async function (global, config) {
     }
   }
 
-  // If no VR devices are present, return a Cardboard device even
-  // if we aren't on mobile so that inline WebXR sessions are at least
-  // supported.
+  // If no WebVR devices are present, check to see if a Cardboard device is
+  // allowed and if so return that.
   // TODO: This probably requires more changes to allow creating an
   // immersive session in a headset that gets connected later.
+  let mobile = isMobile(global);
+  if ((mobile && config.cardboard) ||
+      (!mobile && config.allowCardboardOnDesktop)) {
```

Comment:
Should the desktop mode also require `config.cardboard`? I know that these flags are not ideal :cry: -- although I guess this is where one would configure overriding a native inline device with cardboard one..

Reply:
Agreed it's a bit weird. I was going off of the comments on the config, though:

```js
// Whether a CardboardXRDevice should be discoverable if on
// a mobile device, and no other native (1.1 VRDisplay if `webvr` on,
// or XRDevice) found.
cardboard: true,

// Whether a CardboardXRDevice should be created if no WebXR API found
// on desktop or not. Stereoscopic rendering with a gyro often does not make sense on desktop, and probably only useful for debugging.
allowCardboardOnDesktop: false,
```